### DISCUSSION TITLE
[Question] Does area lights' eval() function take into account of the forshortening factor?

### DISCUSSION THREAD

**Discussion by uguuuuuu** [mitsuba-renderer/mitsuba3#515] [Q&A] [answered]
The documentation says eval() returns radiance along the direction stored in the surface interaction, but I failed to find in the source code where the radiance is multiplied by the forshortening factor cosine theta.

**Comment by ziyi-zhang** [comment, marked as answer]
Hi @uguuuuuu 
`eval` simply queries the textured radiance. The ratio of area measure and solid angle measure is included in the [density term](https://github.com/mitsuba-renderer/mitsuba3/blob/0e497163e13a27d7afe5beb591f6fd7bac7ce80d/src/render/shape.cpp#L365-L367).

**Reply by uguuuuuu** [reply]
Hi @ziyi-zhang, thank you for answering my question!
I believe textured radiance is meant to be the radiance along the direction normal to the surface. Then to go from normal radiance to radiance along another direction, the cosine of the angle made by the normal direction and outgoing direction has to be multiplied. So if eval() just returns the textured radiacne, then it is not doing what the documentation says it is supposed to do. Am I right?

**Reply by ziyi-zhang** [reply]
Apologies. I misunderstood the question. I was talking about the foreshortening term and the Jacobian of a shading point, but your question is actually about the energy (or flux) emitted from the light itself.

For emitters, `eval(si)` is supposed to return the **radiance** (flux per unit area per unit solid angle), not **intensity** (flux per solid angle).
- For a point light, we specify its **intensity**. `eval(si)` is not defined here since there is no surface. But we can query its radiance through `eval_direction(shading_pt)`. The radiance will be the intensity divided by squared distance.
- For an area emitter, we specify its spatially-varying **radiance**. We assume that for any given point on the area light, the emission profile is "uniform" in the sense that the radiance (not flux) along any direction is the same. This assumption can be intuitively understood as, if a perspective camera looks at a rectangle area light with radiance 1, then every pixel would have value 1. `eval(si)` just queries the bilinear interpolation of the provided texture in this case.
- For envmap, we also specify the radiance.
- For a directional emitter, we specify its **irradiance** (flux per unit area).

**Reply by uguuuuuu** [reply]
Thank you Ziyi, that is quite a lesson on radiometry! I didn't know the assumption that emission profiles are uniform for points on area lights and that was why I was so confused, but I think I understand now. Thank you Ziyi, and hope this will help other people having similar questions!
